### PR TITLE
Search endpoint required to find person id from email

### PR DESCRIPTION
**Pull request by msaurabhee**
Please approve this PR and publish the build for the new version,
so that I can use this endpoint.

Thanks.

### CODE REVIEW

**Comment by msaurabhee** [comment]
Can some one approve and merge this? Anyone? What does a guy have to do to get his PR merged around here?

**Review by anthonator — CHANGES_REQUESTED** [review]

File: lib/klaviyo/helpers/url.ex [hunk 1]
```diff
@@ -37,6 +37,10 @@ defmodule Klaviyo.Helpers.URL do
     Map.put(uri, :query, URI.encode_query(operation.params))
   end
 
+  defp put_query(uri, %{ method: :put } = operation, _config) do
```
[the remark this answers is not in the document]
Comment (msaurabhee) [reply lib/klaviyo/helpers/url.ex:40]: Sorry, I ran a request with `:PUT`, it is not working here. So while debugging, I found that the `url` field was not being evaluated with `api_key` parameter. After this modification, PUT started working.

I guess, klaviyo servers expect `api_key` as a parameter in PUT requests.

[the remark this answers is not in the document]
Comment (msaurabhee) [reply lib/klaviyo/helpers/url.ex:40]: @anthonator can you try any `PUT` requests with current master against actual Klaviyo subscription? I was trying against this function:

```
  @doc """
  Update member
  """
  def update_member(person_id, attrs, api_key) do
    with {:ok, response} <-
           Klaviyo.Person.update(person_id, attrs)
           |> Klaviyo.request(%{private_api_key: api_key}) do
      person = Map.get(response, :body)
      {:ok, person}
    end
  end
```

The test is as follows:
```
  describe "update_member" do

    test "update existing member" do
      api_key = "replace_with_api_key"
      person_id = "replace_with_profile_id"
      attrs = %{"external_id" => "", "$id" => "some_id"}

      assert {:ok, %{"id" => ^person_id}} =
               KlaviyoAdapter.update_member(person_id, attrs, api_key)
    end
  end
```

[the remark this answers is not in the document]
Comment (anthonator) [reply lib/klaviyo/helpers/url.ex:40]: Would you mind making this a separate pull request? This seems like a separate concern and I'd like to get `Klaviyo.People.search/1` released for you.

[the remark this answers is not in the document]
Comment (msaurabhee) [reply lib/klaviyo/helpers/url.ex:40]: There you go @anthonator https://github.com/malomohq/klaviyo-elixir/pull/44/files

[the remark this answers is not in the document]
Comment (msaurabhee) [reply lib/klaviyo/helpers/url.ex:40]: @anthonator can you try out the PUT requests as described in the above request against master, and let me know, if this PR can be merged or not? The `PUT` requests don't work as per me.

[the remark this answers is not in the document]
Comment (msaurabhee) [reply lib/klaviyo/helpers/url.ex:40]: @anthonator any updates on this, please?

[the remark this answers is not in the document]
Comment (msaurabhee) [reply lib/klaviyo/helpers/url.ex:40]: @anthonator any updates on this, please (2)?

**Comment by anthonator** [comment]
This should be handled by the most recent rewrite.

**Pull request closed without merging**